### PR TITLE
Update README_template.md

### PR DESCRIPTION
**Pull request by WalkerKnapp**

### CODE REVIEW

File: readme/README_template.md [hunk 1]
```diff
@@ -83,7 +83,7 @@ The following message frameworks are currently supported out of the box:
 * [Javacord](#javacord)
 * [JDA](#jda)
 
-If you want to have support for an additional, do not hesitate to open a pull request or feature request issue.
+If you want to have support for an additional framework, do not hesitate to open a pull request or feature request issue.
```
Comment (Vampire) [inline readme/README_template.md]:
```suggestion
If you want to have support for an additional framework, do not hesitate to open a pull request or feature request
issue.
```

File: readme/README_template.md [hunk 2]
```diff
@@ -222,53 +222,50 @@ public class PingCommand implements Command<Message> {
...
 #### Command Aliases
 
-By overwriting the `Command#getAliases()` method or applying one or multiple [`@Alias`][@Alias JavaDoc] annotations the
-aliases to which the command reacts can be configured. If at least one alias is configured, only the explicitly
-configured ones are available. If no alias is configured, the class name, stripped by `Command` or `Cmd`
-suffix and / or prefix and the first letter lowercased is used as default.
+By overwriting the `Command#getAliases()` method or applying one or multiple [`@Alias`][@Alias JavaDoc] annotations, the
+aliases to which the command reacts can be configured. If no aliases are configured, the class name, with the `Command`
+or `Cmd` suffix/prefix stripped and the first letter lowercased is used as a default. If at least one alias is configured,
+only the explicitly configured ones are used.
```
Comment (Vampire) [inline readme/README_template.md]:
```suggestion
or `Cmd` suffix / prefix stripped and the first letter lowercased is used as a default. If at least one alias is
configured, only the explicitly configured ones are used.
```

File: readme/README_template.md [hunk 3]
```diff
@@ -222,53 +222,50 @@ public class PingCommand implements Command<Message> {
...
 
 #### Asynchronous Command Execution
 
-By overwriting the `Command#isAsynchronous()` method or applying the [`@Asynchronous`][@Asynchronous JavaDoc] annotation
+By overwriting the `Command#isAsynchronous()` method or applying the [`@Asynchronous`][@Asynchronous JavaDoc] annotation,
 the command handler can be told to execute the command asynchronously.
```
Comment (Vampire) [inline readme/README_template.md]:
```suggestion
By overwriting the `Command#isAsynchronous()` method or applying the [`@Asynchronous`][@Asynchronous JavaDoc]
annotation, the command handler can be told to execute the command asynchronously.
```

File: readme/README_template.md [hunk 4]
```diff
@@ -222,53 +222,50 @@ public class PingCommand implements Command<Message> {
 With everything else using the default, this is already enough to have a working ping bot.
 A fully self-contained example can be found at `examples/simplePingBotJavacord`.
 
-To further customize the behavior of a command you can either annotate the command class or overwrite the according
-methods in your command implementation to replace the default implementations which evaluate the annotations.
-Having annotations applied and at the same time overwriting the according methods makes only sense if you want the
-annotations only for documentary purpose or evaluate them yourself, as the default implementations of those methods
-are the only places where the annotations are evaluated by default. Every other place like for example command handler
-call the methods on the command implementation.
+To further customize the behavior of a command, you can either annotate the command class or overwrite the
```
Comment (Vampire) [inline readme/README_template.md]:
```suggestion
To further customize the behavior of a command, you can either annotate the command class or overwrite the
```

File: readme/README_template.md [hunk 5]
```diff
@@ -222,53 +222,50 @@ public class PingCommand implements Command<Message> {
...
 being configured asynchronously if the underlying message framework dispatches message events on different threads.
 
 #### Command Description
 
-By overwriting the `Command#getDescription()` method or applying the [`@Description`][@Description JavaDoc] annotation
-the description of the command can be configured. Currently this description is used nowhere, but can for example be
-displayed in an own help command.
+By overwriting the `Command#getDescription()` method or applying the [`@Description`][@Description JavaDoc] annotation,
+the description of the command can be configured. This description is not currently used by default, but it can be used, 
+for example, in a custom help command.
```
Comment (Vampire) [inline readme/README_template.md]:
```suggestion
the description of the command can be configured. This description is not currently used by this framework, but it can
be used, for example, in a custom help command.
```

File: readme/README_template.md [hunk 6]
```diff
@@ -296,8 +293,9 @@ public class MyFancyServer extends ServerJavacord {
 
 #### Command Usage
 
-By overwriting the `Command#getUsage()` method or applying the [`@Usage`][@Usage JavaDoc] annotation the usage of the
-command can be configured. This usage can for example be displayed in an own help command.
+By overwriting the `Command#getUsage()` method or applying the [`@Usage`][@Usage JavaDoc] annotation, the usage of the
+command can be configured. This usage is not currently used by default, but it can be used, for example, in a custom
+help command.
```
Comment (Vampire) [inline readme/README_template.md]:
```suggestion
command can be configured. This usage can be used, for example, in a custom help command.
```
Whether it is used or not depends on whether the parameter parser is used or not.
I intentionally did not write here that it is not used opposed to the description which is really not used by the framework currently, as it **is** used under certain circumstances.

File: readme/README_template.md [hunk 7]
```diff
@@ -222,53 +222,50 @@ public class PingCommand implements Command<Message> {
...
 
-To further customize the behavior of a command you can either annotate the command class or overwrite the according
-methods in your command implementation to replace the default implementations which evaluate the annotations.
-Having annotations applied and at the same time overwriting the according methods makes only sense if you want the
-annotations only for documentary purpose or evaluate them yourself, as the default implementations of those methods
-are the only places where the annotations are evaluated by default. Every other place like for example command handler
-call the methods on the command implementation.
+To further customize the behavior of a command, you can either annotate the command class or overwrite the 
+corresponding methods. Annotations are ignored when the corresponding methods are overwritten, but they can still be
+separately evaluated or used as documentation.
```
Comment (Vampire) [inline readme/README_template.md]:
```suggestion
separately evaluated or used as documentation. The default implementations of those methods are the only locations
where the annotations are evaluated by default. Every other place calls the methods on the command implementation.
```
Imho the new text is too condensed.
What do you think about re-adding these two sentences to the end?
Probably cleaned from errors of course :-D

File: readme/README_template.md [hunk 8]
```diff
@@ -475,23 +473,23 @@ The five phases that are handled are in order:
 For all but the first and last, there is a before and an after sub phase each during which the command context
 transformer is called.
 
-If at the end of the initialization phase, any before, or any after sub phase the command is set
+If at the end of the initialization phase, or any before/after sub phases, the command is set
```
Comment (Vampire) [inline readme/README_template.md]:
```suggestion
If at the end of the initialization phase, or any before / after sub phases, the command is set
```

File: readme/README_template.md [hunk 9]
```diff
@@ -475,23 +473,23 @@ The five phases that are handled are in order:
...
 in the context, processing is fast forwarded immediately to the command execution phase and all other
 inbetween phases and sub phases are skipped.
 
-If at the end of the initialization phase, any before, or any after sub phase before the
+If at the end of the initialization phase, or any before/after sub phases before the
```
Comment (Vampire) [inline readme/README_template.md]:
```suggestion
If at the end of the initialization phase, or any before / after sub phases before the
```

File: readme/README_template.md [hunk 10]
```diff
@@ -475,23 +473,23 @@ The five phases that are handled are in order:
...
 `BEFORE_COMMAND_COMPUTATION` sub phase, the alias is set in the context, processing is
 fast forwarded immediately to the before command computation sub phase and all other inbetween
 phases and sub phases are skipped.
 
-If at the end of the initialization phase, any before, or any after sub phase before the
+If at the end of the initialization phase, any before/after sub phases before the
```
Comment (Vampire) [inline readme/README_template.md]:
```suggestion
If at the end of the initialization phase, or any before / after sub phases before the
```

File: readme/README_template.md [hunk 11]
```diff
@@ -538,19 +536,18 @@ a command not found event is being fired and processing stops completely.
...
+For each of the described sub phases, exactly one context transformer that is compatible with the framework message
 can be registered. If you need multiple such transformers, then make one distributing transformer that calls the
 other transformers in the intended order. You can also register one context transformer for multiple phases.
 The `transform` method gets the current phase as argument and can then decide what to do based on that phase parameter.
 
 A command context transformer can be registered by providing a CDI bean that implements the
 [`CommandContextTransformer`][CommandContextTransformer JavaDoc] interface. In the implementation of the `transform`
-method you can determine from the current command context and the given phase how you like the command context to
-be and return the adapted command context or even a completely new command context.
+method, you can transform current command context given a phase, or return a completely new command context.
```
Comment (Vampire) [inline readme/README_template.md]:
```suggestion
method, given the current phase, you can transform the current command context, or even return a completely new one.
```

File: readme/README_template.md [hunk 12]
```diff
@@ -538,19 +536,18 @@ a command not found event is being fired and processing stops completely.
...
 Additionally the bean has to be annotated with at least one [`@InPhase`][@InPhase JavaDoc] qualifier annotation, or it
 will simply not be used silently. The transformer will be called for each sub phase that is added with that annotation.
 
-There are also helper classes that can be used as super classes for own command context transformers like for example
+There are also helper classes that can be used as super classes for own command context transformers, for example,
 a provider that returns the mention string for the bot as command prefix if Javacord is used as the underlying
```
Comment (Vampire) [inline readme/README_template.md]:
```suggestion
a transformer that returns the mention string for the bot as command prefix if Javacord is used as the underlying
```

**Review by Vampire — CHANGES_REQUESTED** [review]
Thank you very much, I have some annotations and suggestions though. Mind quickly looking through them? :-)

File: readme/README_template.md (hunk 7 above)
Comment (WalkerKnapp) [reply readme/README_template.md]: Sure, here is a way you could reword it to flow a little bit better, let me know what you think

```suggestion
separately evaluated or used as documentation. For all functionality (i.e. command handlers),
command-framework uses the command method implementations. Annotations are only read
in the default implementations of those methods.
```

File: readme/README_template.md (hunk 6 above)
Comment (WalkerKnapp) [reply readme/README_template.md]: Ah, I see. I just moved this phrasing from the `Command Description` section.

File: readme/README_template.md (hunk 7 above)
Comment (Vampire) [reply readme/README_template.md]: i.e. is not correct, is it?
That would mean all functionality is in the command handlers only.
How about
```suggestion
separately evaluated or used as documentation. For all functionality this framework uses the command method
implementations. The annotations are only read in the default implementations of those methods.
```

Comment (WalkerKnapp) [reply readme/README_template.md]: Oh, yes, it should be `i.g.` instead. But you're right that it is a bit unnecessary. That last version lgtm!

**Comment by Vampire** [comment]
Thank you very much, I picked it manually to the master branch. :-)
